.gitignore

```diff
@@ -11,3 +11,53 @@
 !project.code-workspace
 
 # Please, add your custom content below!
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+.yarn/install-state.gz
```

Comment:
it looks like parts of the repository are written like we're using yarn and parts are written like we are using npm - would be good to pick a package manager and stick with it